src/crypto/PublicKey.ts

```diff
@@ -225,7 +225,18 @@ export class PublicKey
       secp256k1Uncompressed: obj.secp256k1Uncompressed,
     })
     if (obj.signature) {
-      this.signature = new Signature(obj.signature)
+      // Handle a case where Flutter was publishing signatures with walletEcdsaCompact
+      // instead of ecdsaCompact for v1 keys.
+      if (!obj.signature.ecdsaCompact && obj.signature.walletEcdsaCompact) {
```

Comment:
Do we need to do this if we expand the getPublicKey functionality below?

Reply:
It's a good question! It feels like a good move to fix the signature here just in case it propagates elsewhere and ends up serialized, but analyzing that seems tricky.

Reply:
Yeah, I was worried about the signature assumption being made elsewhere (even in the future) and thought the best fix might be fixing the PublicKey object itself. It's a best attempt at least imo.

Reply:
I feel safer leaving the check in both places because I can't guarantee the new constructor will be called in all cases. I know it's a little messier, but I confirmed this does solve the issue at hand with a test wallet. I think I'm going to merge as-is unless you strongly disagree!